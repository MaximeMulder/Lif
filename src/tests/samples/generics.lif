function super_print[T](value: T) {
    print(value);
}

super_print[Integer](42);
super_print[String]("Forty-two");

class Queue[T] {
    function push(self, element: T) {
        self.elements.append(element);
    }
}

let queue: Queue[Integer] = new(Queue[Integer]);
queue.elements = Array[Integer]();
queue.push(0);
queue.push(1);
print(queue.elements(0));
print(queue.elements(1));
