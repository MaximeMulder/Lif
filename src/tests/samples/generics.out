42
Forty-two
0
1
